packages/quill/src/modules/clipboard.ts

```diff
@@ -26,10 +26,23 @@ import normalizeExternalHTML from './normalizeExternalHTML/index.js';
 
 const debug = logger('quill:clipboard');
 
-type Selector = string | Node['TEXT_NODE'] | Node['ELEMENT_NODE'];
-type Matcher = (node: Node, delta: Delta, scroll: ScrollBlot) => Delta;
+type TextSelector = Node['TEXT_NODE'];
+type ElementSelector = string | Node['ELEMENT_NODE'];
+type Selector = TextSelector | ElementSelector;
 
-const CLIPBOARD_CONFIG: [Selector, Matcher][] = [
+type TextMatcher = (node: Text, delta: Delta, scroll: ScrollBlot) => Delta;
```

Comment:
Looks like in some cases this change makes things a bit harder. E.g. https://share.slab.com/2dv3PhtG, previously, the parameters have the correct typing, but with this change, TypeScript can't infer the types and require users to provide.

Reply:
You're correct -- inference isn't really possible and that was actually intentional.  The goal of the PR was that the matchers need to be strongly typed to prevent misconfiguration.  

In your example, previously the parameters would have just simply been `(node: Node, ...)`.  While it is inferring the signature, it isn't protecting against mismatch.

Is type inference a requirement here?  If so, then I would say the PR should be closed.

Reply:
Okay no problem, then probably this PR would do more harm than good so I'll close it out!